### PR TITLE
Migrate `remap-path-prefix-dwarf` `run-make` test to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Possibly my proudest branch name yet.

try-job: aarch64-apple
try-job: armhf-gnu
try-job: test-various
try-job: x86_64-gnu-llvm-18

### CODE REVIEW

**Comment by Oneirical** [comment]
Ignore restored, reason added. @rustbot review

File: tests/run-make/remap-path-prefix-dwarf/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+//@ ignore-windows
+// Reason: the remap path prefix is not printed in the dwarf dump.
+
+use run_make_support::{cwd, is_darwin, llvm_dwarfdump, rust_lib_name, rustc};
+
+fn main() {
+    // The compiler is called with an *ABSOLUTE PATH* as input, and that absolute path *is* within
+    // the working directory of the compiler. We are remapping the path that contains `src`.
+    check_dwarf(DwarfTest {
```
Comment (jieyouxu) [inline tests/run-make/remap-path-prefix-dwarf/rmake.rs:15]: Remark: this organization is very nice :+1:

**Review by jieyouxu — APPROVED** [review]
Nice

**Comment by jieyouxu** [comment]
Cool.
@bors r+

**Comment by Oneirical** [comment]
Rebased. Please re-approve. @rustbot review

**Pull request merged**